### PR TITLE
refactor gtd code and tests

### PR DESCRIPTION
**Pull request by max397574**

### CODE REVIEW

File: tests/core/gtd/queries/helpers_spec.lua [hunk 1]
```diff
@@ -47,7 +47,44 @@ describe("CORE.GTD.QUERIES - Helpers:", function()
...
+        assert.equals(
+            queries_helper.date_converter("12mon"),
+            os.date("%Y-%m-%d", get_date_in_x_days(84 - values[os.date("%A")]))
+        )
+
+        assert.is_nil(queries_helper.date_converter("9abc"))
+
+        assert.equals(
+            queries_helper.date_converter("mon"),
+            os.date("%Y-%m-%d", get_date_in_x_days(7 - values[os.date("%A")]))
```
Comment (danymat) [inline tests/core/gtd/queries/helpers_spec.lua:86]: This will not work on every machine, I think you have to reconsider this test.

Comment (joaothallis) [reply tests/core/gtd/queries/helpers_spec.lua:86]: @danymat why it will not work on some machine?

Comment (max397574) [reply tests/core/gtd/queries/helpers_spec.lua:86]: it depends on the language

**Comment by max397574** [comment]
I'll open a new pr for this since it has merge conflicts and I lost the branch somehow

**Pull request closed without merging**